README.md

```diff
@@ -1237,3 +1237,32 @@ DELETE /api/v1/users/mary
 ```
 HTTP/1.1 200 OK
 ```
+
+# Upgrades
+
+Initiates an upgrade to the provided version. Once this endpoint is called [Horticulturalist](https://github.com/medic/horticulturalist) will take over and complete the installation.
+
+### Permissions
+
+`_admin` (CouchDB Administrator)
+
+## Example
+
+```
+POST /api/v1/upgrade
+{
+  "build": {
+    "namespace": "medic",
+    "application": "medic",
+    "version": "3.0.0-beta.1"
+  }
+}
+```
+
```

Comment:
I would agree with you if it didn't make the code more complex because  of handling and documenting namespace and app name in all the places.  Seems more logical to me to keep it more simple and flexible as a first cut and lock it down further later when these needs become more clear.  But as I look at it more it is growing on me.